File: recursion3/kpc.cpp
#include<iostream>
#include<string>
#include<vector>
using namespace std;

string codes[10] = {
    ".",
    "abc",
    "def",
    "ghi",
    "jklm",
    "no",
    "pqrs",
    "tu",
    "vwxy",
    "z"
};

vector<string> getkpc(string str){
    
     if(str.length()==0)
     {
     vector<string> bres;
     bres.push_back(" ");
     return bres;
    }

    char ch = str[0];//8
    string ros = str.substr(1,str.length()-1);//35
    vector<string> rres  = getkpc(ros);//35 all combinations #faith
    vector<string> mres; //resultant vector

    string code = codes[ch-48]; //letters for 8

    for(int i =0 ; i<code.size() ; i++){
        //take first letter of 8 and add it to all the combinatiions of 35
        //repeat for all letters of 8
        char chcode = code[i];

        for(int j =0  ; j<rres.size() ; j++){
            string rstrr = rres[j];
            mres.push_back(chcode+rstrr);

        }
            }
            return mres;

}

int main(int argc , char** argv){
    vector<string> result = getkpc("835");
    int counter = 0;
    for(int i =0 ; i<result.size() ; i++){
        cout<<result[i]<<endl;
        counter++;
            }
            cout<< counter;
}

File: recursion5/kpc.java
import java.util.*;
public class kpc{

           static String[] codes= {
    ".",
    "abc",
    "def",
    "ghi",
    "jklm",
    "no",
    "pqrs",
    "tu",
    "vwxy",
    "z"
};


public static void pkpc(String q , String asf){
 
        if(q.length()==0){
            System.out.println(asf);
            return;
        }    

        char ch = q.charAt(0);
        String ros = q.substring(1,q.length());

        String chcode = codes[ch-48];
        for(int i = 0 ; i<chcode.length() ; i++){
            pkpc(ros , asf + chcode.charAt(i));
        }

}
        public static void main(String[] args){

            pkpc("835" , " ");

    }
}